rootex/framework/systems/pause_system.cpp

```diff
@@ -0,0 +1,38 @@
+#include "pause_system.h"
+
+#define LINE_MAX_VERTEX_COUNT 1000
+#define LINE_VERTEX_COUNT 2
+
+PauseSystem* PauseSystem::GetSingleton()
+{
+	static PauseSystem singleton;
+	return &singleton;
+}
+
+PauseSystem::PauseSystem()
+    : System("PauseSystem", UpdateOrder::Async, true)
+{
+	m_IsPausingEnabled = true;
+}
+
+void PauseSystem::update(float deltaMilliseconds)
+{
+	Scene* root = SceneLoader::GetSingleton()->getCurrentScene();
+	if (m_IsPausingEnabled)
+	{
+		if (!(Scene::FindScenesByName("PauseScene").size()))
+		{
+			Ptr<Scene>& pauseScene = Scene::CreateFromFile("game/assets/scenes/pause.scene.json");
+			pauseScene->setName("PauseScene");
+			root->addChild(pauseScene);
+		}
+	}
+	else
+	{
+		Vector<Scene*> pauseScenes = Scene::FindScenesByName("PauseScene");
```

Comment:
Open an issue to create a mechanism for Reserved Entity Names, so that a player won't be able to create entities that interfere with internal engine mechanisms